### PR TITLE
No more useless subst

### PR DESCRIPTION
**Pull request by HoshinoTented**
The _subst of parameters of Ctor_ now disappear after tyck that `Pat.Ctor`. For example:

```aya
open data Nat
| O
| S (_1 : Nat)

def justMatch (m n : Nat) : Nat
| O, n => O
| S m', n => m'
```

There is a subst `_1 |-> m'` when we tyck the  `n` pattern in `| S m', n => m'`, which is useless.

### CODE REVIEW

File: base/src/main/java/org/aya/tyck/pat/PatTycker.java [hunk 1]
```diff
@@ -59,21 +59,30 @@ public final class PatTycker {
...
    * An {@code as pattern} map.
-   * Every binding in the function telescope was treated as an {@code as pattern}
-   * so they were added to this map.
    */
   private final @NotNull TypedSubst patSubst;
+
+  /**
+   * Every binding in the function telescope was treated as an {@code as pattern},
+   * but they won't be added to {@link PatTycker#patSubst},
+   * because we may visit a complete different signature during tycking {@link Pat.Ctor},
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/pat/PatTycker.java:68]: completely

File: base/src/main/java/org/aya/tyck/pat/PatTycker.java [hunk 2]
```diff
@@ -59,21 +59,30 @@ public final class PatTycker {
...
+   * and the substs for that signature become useless after tycking {@link Pat.Ctor}
+   * (Also, the substs for the function we tycking is useless when we tycking the {@link Pat.Ctor}).
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/pat/PatTycker.java:70]: the function we are tycking... when we tyck the...

File: base/src/main/java/org/aya/tyck/pat/PatTycker.java [hunk 3]
```diff
@@ -320,7 +333,8 @@ yield doTyck(new Pattern.FakeShapedList(
   public @NotNull Tuple2<SeqView<Pat>, Term>
   visitPatterns(@NotNull Def.Signature sig, @NotNull SeqView<Pattern> stream, @Nullable Pattern outerPattern) {
     var results = MutableList.<Pat>create();
-    if (sig.param().isEmpty() && stream.isEmpty()) return Tuple.of(results.view(), sig.result());
+    // this path seems useless
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/pat/PatTycker.java:336]: Is it still reachable?

Comment (ice1000) [reply base/src/main/java/org/aya/tyck/pat/PatTycker.java:336]: I mean like does the code coverage shows hit for this line

Comment (HoshinoTented) [reply base/src/main/java/org/aya/tyck/pat/PatTycker.java:336]: It is reachable, but the behavior won't change if we delete that line:

* `sig.param().isEmpty`: won't go through the `while` loop
* `stream.isEmpty()`: won't go through the `if` test

Comment (ice1000) [reply base/src/main/java/org/aya/tyck/pat/PatTycker.java:336]: Oh

**Pull request merged**